File: api/src/main/java/com/tainzhi/sample/api/adapter/BasicHorizontalAdapter.java
package com.tainzhi.sample.api.adapter;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.tainzhi.sample.api.R;

/**
 * Created by muqing on 2019-10-08.
 * Email: qfq61@qq.com
 */
public class BasicHorizontalAdapter extends BaseQuickAdapter<String, BaseViewHolder> {
	public BasicHorizontalAdapter() {
		super(R.layout.item_basic_list);
	}
	
	@Override
	protected void convert(@NonNull BaseViewHolder helper, String item) {
		helper.setText(R.id.activity_name, item);
	}
}

File: module-customview/src/main/java/com/tainzhi/sample/customview/widget/BasicHorizontalAdapter.kt
package com.tainzhi.sample.customview.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.tainzhi.sample.customview.R

/**
 * Created by muqing on 2019-10-08.
 * Email: qfq61@qq.com
 */
class BasicHorizontalAdapter :
    BaseQuickAdapter<NameClass, BaseViewHolder>(
        R.layout.item_basic_list
    ) {
    override fun convert(holder: BaseViewHolder, item: NameClass) {
        holder.setText(R.id.activity_name, item.className)
    }
}

data class NameClass(val className: String, val clazz: Class<*>)